src/tests/ftest/util/storage_utils.py

```diff
@@ -0,0 +1,629 @@
+            # Sort the detected devices and place then in lists by NUMA node
+            numa_devices = self._get_numa_devices(self.controller_devices or self.disk_devices)
+            self._log.debug('  NVMe/VMD numa_devices:   %s', numa_devices)
+
+            # Interleave the devices for bdev_list distribution.
+            interleaved = self._get_interleaved(engines, numa_devices)
+            self._log.debug('  NVMe/VMD interleaved:    %s', interleaved)
+
+            # Break the interleaved (prioritized) disk list into groups of engine size
+            device_sets = [interleaved[x:x + engines] for x in range(0, len(interleaved), engines)]
+            self._log.debug('  NVMe/VMD device_sets:    %s', device_sets)
+
+            # Get the tier number device placement for the available number of devices
+            tier_placement = []
+            while len(tier_placement) < len(device_sets):
+                tier_placement.extend(self.TIER_NVME_PRIORITY[:max_nvme_tiers])
+            tier_placement = sorted(tier_placement[:len(device_sets)])
+            self._log.debug('  NVMe/VMD tier_placement: %s', tier_placement)
+            tiers += max(tier_placement)
+
+            bdev_list = {}
+            for device_set in device_sets:
+                tier = tier_placement.pop(0)
+                for engine, device in enumerate(device_set):
+                    if engine not in bdev_list:
+                        bdev_list[engine] = {}
+                    if tier not in bdev_list[engine]:
+                        bdev_list[engine][tier] = []
+                    bdev_list[engine][tier].append(f'"{device}"')
+            self._log.debug('  NVMe/VMD bdev_list:      %s', bdev_list)
+
+        lines = ['server_config:', '  engines:']
+        for engine in range(engines):
+            lines.append(f'    {str(engine)}:')
+            lines.append('      storage:')
+            for tier in range(tiers):
+                lines.append(f'        {str(tier)}:')
+                if tier == 0 and pmem_list:
+                    lines.append('          class: dcpm')
+                    lines.append(f'          scm_list: ["{pmem_list[engine]}"]')
+                    lines.append(f'          scm_mount: /mnt/daos{engine}')
+                elif tier == 0:
+                    lines.append('          class: ram')
+                    lines.append(f'          scm_mount: /mnt/daos{engine}')
+                    lines.append(f'          scm_size: {scm_size}')
+                else:
+                    lines.append('          class: nvme')
+                    lines.append(f'          bdev_list: [{", ".join(bdev_list[engine][tier])}]')
+
+        self._log.debug('  Creating %s', yaml_file)
+        for line in lines:
+            self._log.debug('    %s', line)
+        try:
+            with open(yaml_file, "w", encoding="utf-8") as config_handle:
+                config_handle.writelines([f'{line}\n' for line in lines])
```

Comment:
FYI that with a very large list, this is potentially faster (generator instead of list), and works just fine on small lists too
```suggestion
                config_handle.writelines(f'{line}\n' for line in lines)
```